### PR TITLE
text input/output and tables

### PR DESCRIPTION
**Pull request by petrelharp**
Here's a start at #498 .

Note the bugfix in TableCollection.__str__().

### CODE REVIEW

File: docs/interchange.rst [hunk 1]
```diff
@@ -636,6 +636,36 @@ ignored (IDs are always determined by the position of the row in a table).
...
+Note that there are currently no globally defined ``flags``, but the column
+is still required; a value of ``0`` means that there are no flags set.
+
+The ``location`` column should be a sequence of comma-separated numeric
+values. They do not all have to be the same length.
+
+An example individual table::
+
+    flags   location
+    1           0.5,1.2
```
Comment (jeromekelleher) [inline docs/interchange.rst]: Is it confusing to set these to 1 when we have no flags defined?

File: docs/interchange.rst [hunk 2]
```diff
@@ -721,6 +751,21 @@ mutations::
...
+
+Population tables only have a ``metadata`` column, so the text format for
+a population table requires there to be a ``metadata`` column. See the
+:ref:`population table definitions <sec_population_table_definition>` for
+details.
+
+An example population table::
+
+    id   metadata
+    0    pop1
```
Comment (jeromekelleher) [inline docs/interchange.rst]: This might be misleading, as the metadata will be base64 decoded.

**Comment by jeromekelleher** [comment]
Do we want to get this one in for 0.6.1 @petrelharp? I'm happy to let it slide until afterwards, personally, as it's not really on any critical paths for me.

**Comment by petrelharp** [comment]
I've added tests, rebased, and squashed; should be good to go?

**Comment by jeromekelleher** [comment]
LGTM, thanks @petrelharp!

**Pull request merged**